src/main/java/de/uni_kl/cs/discodnc/bounds/disco/pwaffine/LeftOverService.java

```diff
@@ -158,15 +158,37 @@ public static ServiceCurve fifoMux(ServiceCurve service_curve, ArrivalCurve arri
      * @return The FIFO service curve
      */
     public static ServiceCurve arbMux(ServiceCurve service_curve, ArrivalCurve arrival_curve) {
-        if (arrival_curve.equals(Curve.getFactory().createZeroArrivals())
-                || service_curve.equals(Curve.getFactory().createZeroDelayInfiniteBurst())) {
-            return service_curve.copy();
-        }
-        if (service_curve.equals(Curve.getFactory().createZeroService())) {
-            return Curve.getFactory().createZeroService();
-        }
-
-        return Curve.getFactory().createServiceCurve(
-                Curve.boundAtXAxis(Curve.sub(service_curve, arrival_curve)));
+    	Pair<Boolean,ServiceCurve> special_cases = computeSpecialValues(service_curve, arrival_curve);
+    	
+    	if(special_cases.getFirst().booleanValue() == true) {
+    		return special_cases.getSecond(); 
+    	} else {
+            return Curve.getFactory().createServiceCurve(Curve.boundAtXAxis(Curve.sub(service_curve, arrival_curve)));
+    	}
+    }
+    
+    private static Pair<Boolean,ServiceCurve> computeSpecialValues(ServiceCurve service_curve, ArrivalCurve arrival_curve) {
+    	ServiceCurve ZERO_SERVICE = Curve.getFactory().createZeroService();
+    	ServiceCurve INFINITE_SERVICE = Curve.getFactory().createZeroDelayInfiniteBurst();
+    	
+    	ArrivalCurve ZERO_ARRIVALS = Curve.getFactory().createZeroArrivals();
+    	ArrivalCurve INFINITE_ARRIVALS = Curve.getFactory().createInfiniteArrivals();
+    	
+    	if(arrival_curve.equals(INFINITE_ARRIVALS)
```

Comment:
We should check always against the constant: INIFINITE_ARRIVALS.equals(arrival_curve) - can avoid null pointer exceptions

Reply:
Could you please correct this too? It can lead to NPE.

Reply:
Sure, I checked the code. It first does a ``obj instanceof class`` where obj is object given to the equals method and class is the class of the object whose equals method is called. ``null instanceof class`` returns false. Thus, we can prevent the NPE by simply swapping to order to ``INFINITE_ARRIVALS.equals(arrival_curve)`` (same for other equals invocations). Done in commit 26b4a13297d4d6631f6bd7d785d970a865a48728